tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -457,16 +457,19 @@ public async Task HandlePlaybackRequest(string recordingId, HttpRequest incoming
 
             var sanitizers = SanitizerRegistry.GetSanitizers(session);
 
-            // we don't need to re-sanitize with recording-applicable sanitizers every time. just the very first one
-            lock (session)
+            if (!session.IsSanitized)
```

Comment:
This is probably safe enough for our purposes, but if we see bugs, or are concerned the memory model may not guarantee safeness, we could replace with `Lazy<T>`.